Connectors/Rhino/Speckle.Connectors.RhinoShared/HostApp/RhinoLayerBaker.cs

```diff
@@ -16,6 +16,13 @@ public class RhinoLayerBaker : TraversalContextUnpacker
 {
   private readonly RhinoMaterialBaker _materialBaker;
   private readonly RhinoColorBaker _colorBaker;
+
+  /// <summary>
+  /// The layer cache storing the full name of created layers.
+  /// Full names should be stored in all lowercase due to case-agnostic layer name uniqueness requirements in Rhino.
+  /// TryGetValue on this dict should also test for lowercase-only names.
+  /// </summary>
+  /// <remarks>The case-agnostic requirement applies to some models (eg Revit with linked models) that may have multiple collections with the same name but with different capitalizations.</remarks>
   private readonly Dictionary<string, int> _hostLayerCache = new();
```

Comment:
Change the StringComparer here to be a case insensitive one rather than doing ToLower() everywhere